swift-extensions/UISwitchExtensions.swift

```diff
@@ -0,0 +1,72 @@
+import UIKit
+import TRIKOT_FRAMEWORK_NAME
+
+extension UISwitch {
+
+    private struct AssosiatedKeys {
+        static var impactFeedbackStyle = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+    }
+    
+    public var switchViewModel: SwitchViewModel? {
+        get { return trikotViewModel() }
+        set(value) {
+            if impactFeedbackStyle() == nil { impactFeedbackStyle(value: .light) }
+            
+            viewModel = value
+            guard let switchViewModel = value else { return }
+
+            self.addTarget(self, action: #selector(UISwitch.onValueChanged), for: .valueChanged)
+            observe(switchViewModel.isOn) { [weak self] (value: Bool) in
+                self?.isOn = value
+            }
+        }
+    }
+
+    @objc
+    func onValueChanged(sender: UISwitch) {
+        if #available(iOS 10.0, *), let style = impactFeedbackStyle() {
+            UIImpactFeedbackGenerator(style: getFeedbackStyle(style: style)).impactOccurred()
+        }
+        switchViewModel?.changeState(state: self.isOn)
+    }
+
+    private func impactFeedbackStyle() -> FeedbackStyle? {
+        return objc_getAssociatedObject(self, AssosiatedKeys.impactFeedbackStyle) as? FeedbackStyle
+    }
+
+    private func impactFeedbackStyle(value: FeedbackStyle?) {
```

Comment:
setImpactFeedbackStyle?

Reply:
keeping the naming scheme used throughout the app